### PR TITLE
Add kplex v1.4

### PR DESCRIPTION
**Pull request by mabster314**
## Add kplex - a multiplexer for various nmea 0183 interfaces
https://www.stripydog.com/kplex/

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mabster314** [comment]
force pushed after making commits with wrong user.email

File: pkgs/applications/misc/kplex/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  buildPhase = ''
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp kplex $out/bin
+  '';
```
Comment (Janik-Haag) [inline pkgs/applications/misc/kplex/default.nix:21]:
```suggestion
  installPhase = ''
    install -D -m 0555 kplex $out/bin/kplex
  '';
```
we can use install here which makes file permissions a bit more obvious

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/kplex/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  meta = with lib; {
+    description = "A multiplexer for various nmea 0183 interfaces";
+    homepage = "https://www.stripydog.com/kplex/";
+    license = licenses.gpl3;
```
[the remark this answers is not in the document]
Comment (jopejoe1) [reply pkgs/applications/misc/kplex/default.nix:26]: gpl3Plus i this case, if you read the part before the license starts it says 
> either version 3 of the License, or
(at your option) any later version.

File: pkgs/by-name/kp/kplex/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  fetchFromGitHub,
+  lib,
+}:
+stdenv.mkDerivation rec {
+  pname = "kplex";
+  version = "1.4";
+  src = fetchFromGitHub {
+    owner = "stripydog";
+    repo =  "kplex";
+    rev = "v${version}";
```
Comment (h7x4) [inline pkgs/by-name/kp/kplex/package.nix:12]: Consider using the `finalAttrs` pattern. It makes it a bit easier to override the package. (You'll have to add the last parenthesis as well)

```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "kplex";
  version = "1.4";
  src = fetchFromGitHub {
    owner = "stripydog";
    repo =  "kplex";
    rev = "v${finalAttrs.version}";
```

File: pkgs/by-name/kp/kplex/package.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    hash = "sha256-sps9l238hGLJ673kewFH8fOJw0HphEkZbJ+VUIzxC+o=";
+  };
+
+  installPhase = ''
+    install -D -m 0555 kplex $out/bin/kplex
+  '';
```
Comment (h7x4) [inline pkgs/by-name/kp/kplex/package.nix:18]: The install changes are optional, but the hooks should be there.

```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm555 kplex -t $out/bin
    runHook postInstall
  '';
```

File: pkgs/by-name/kp/kplex/package.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "A multiplexer for various nmea 0183 interfaces";
+    homepage = "https://www.stripydog.com/kplex/";
```
Comment (h7x4) [inline pkgs/by-name/kp/kplex/package.nix:22]: A changelog is nice to note down, when it exists.

```suggestion
    homepage = "https://www.stripydog.com/kplex/";
    changelog = "https://www.stripydog.com/kplex/changes.html";
```

**Review by h7x4 — COMMENTED** [review]
Here's a few more comments

File: pkgs/by-name/kp/kplex/package.nix (hunk 3 above)
Comment (mabster314) [reply pkgs/by-name/kp/kplex/package.nix:12]: Does mkDerivation do a fixed-point evaluation of the around finalAttrs? I can't find any documentation for this

Comment (h7x4) [reply pkgs/by-name/kp/kplex/package.nix:12]: Yeah, I think so. It's *relatively* new, but I think the main intention is to not have to override multiple attributes when doing `package.override { version = ... }`. This PR contains a lot of information and relevant links if you're interested: #234651

**Comment by h7x4** [comment]
Everything seems good to me now, apart from the commits. Would you mind squashing them into two commits, with the following messages?

```
maintainers: add mabster314
kplex: init at 1.4
```

**Comment by mabster314** [comment]
The ofborg results are showing a bunch of "no space on device." Is this a backend issue?

**Comment by h7x4** [comment]
Probably yeah. First time I ever see it.

**Comment by h7x4** [comment]
@ofborg eval

**Comment by h7x4** [comment]
@mabster314 not sure how I retrigger eval-lib-tests, maybe you could retrigger it manually by amending the last commit, forcing a timestamp change?

**Comment by K900** [comment]
@ofborg eval

**Comment by mabster314** [comment]
great, looks like it worked that time

**Comment by h7x4** [comment]
@Janik-Haag Could you have another look?

**Comment by mabster314** [comment]
@Janik-Haag Can you please take another look at this? The changes you requested have been addressed

**Review by Janik-Haag — APPROVED** [review]
LGTM, sorry for taking so long, I was sick the last few days so I couldn't do any reviews.

**Pull request merged**